modules/imgaug/src/functional.cpp

```diff
@@ -0,0 +1,76 @@
+#include "precomp.hpp"
+
+namespace cv{
+
+    void adjustBrightness(Mat& img, double brightness_factor){
+        CV_Assert(brightness_factor >= 0);
+
+        int channels = img.channels();
+        if(channels != 1 && channels != 3){
+            CV_Error(Error::BadNumChannels, "Only support images with 1 or 3 channels");
+        }
+        img = img * brightness_factor;
+    }
+
+    void adjustContrast(Mat& img, double contrast_factor){
+        CV_Assert(contrast_factor >= 0);
+
+        int num_channels = img.channels();
+        if(num_channels != 1 && num_channels != 3){
+            CV_Error(Error::BadNumChannels, "Only support images with 1 or 3 channels");
+        }
+       Mat* channels = new Mat[num_channels];
+        split(img, channels);
+        std::vector<Mat> new_channels;
+        for(int i=0; i < num_channels; i++){
+            Mat& channel = channels[i];
+            Scalar avg = mean(channel);
```

Comment:
The function [cv::mean](https://docs.opencv.org/4.x/d2/de8/group__core__array.html#ga191389f8a0e58180bb13a727782cd461) calculates the mean value M of array elements, independently for each channel, and return it. No need to allocate array and iterate over channels the next arithmetic steps could be dome for all channels together without loop.